### ISSUE TITLE
Translate ImmutableHashSet<T>.Contains() and HashSet<T>.Contains()

### ISSUE THREAD

**Issue by ronnygunawan** [dotnet/efcore#18857]
Current EF Core doesn't translate `IImmutableHashSet<T>.Contains()` and `HashSet<T>.Contains()`:
```cs
public async Task<List<Person>> GetMultipleAsync(ImmutableHashSet<int> ids) {
    return await (from p in _dbContext.People
                  where ids.Contains(p.Id)
                  select p).ToListAsync();
}

public async Task<List<Person>> GetMultipleAsync(HashSet<int> ids) {
    return await (from p in _dbContext.People
                  where ids.Contains(p.Id)
                  select p).ToListAsync();
}
```

Please add translation of `IImmutableHashSet<T>.Contains()` and `HashSet<T>.Contains()` to `Query/Internal/ContainsTranslator.cs`

**Comment by ajcvickers** [comment]
@ronnygunawan Thanks for filing this issue. We will discuss in our next triage meeting.

**Comment by smitpatel** [comment]
cc: @roji - did we cover these types already?

**Comment by roji** [comment]
Yes, Contains on any ICollection<T> should be handled by #18634. The fix is going out in 3.1.0-preview3, which should get released in the next couple of days - @ronnygunawan can you please test on it when it's available?

**Comment by roji** [comment]
Duplicate of #17342
